### PR TITLE
[JENKINS-29922] Adding symbols for a smoother step appearance

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-29922](https://issues.jenkins-ci.org/browse/JENKINS-29922)

- [X] update to new parent (incl. FindBugs & `-source 7` cleanups)
- [X] add `@Symbol`
- [X] verify usage in tests
- [x] update demo (needs some plugins released and propagated to UC)

@reviewbybees esp. @abayer, @kohsuke

### CODE REVIEW

File: demo/plugins.txt [hunk 1]
```diff
@@ -12,12 +12,12 @@ parallel-test-executor:1.8
...
-workflow-api:1.13
-workflow-basic-steps:1.13
-workflow-cps:1.13
-workflow-durable-task-step:1.13
-workflow-job:1.13
-workflow-multibranch:1.13
-workflow-scm-step:1.13
-workflow-step-api:1.13
-workflow-support:1.13
+workflow-api:2.0
```
Comment (jglick) [inline demo/plugins.txt]: Versions are approximate, will need to be refined.

File: src/main/resources/org/jenkinsci/plugins/parallel_test_executor/SplitStep/help.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<div>
+    Asks Jenkins to analyze the timing of tests from the last build and divide the tests for this build into roughly equal subsets for <code>parallel</code> execution.
+    <a href="https://github.com/jenkinsci/parallel-test-executor-plugin/blob/master/demo/repo/Jenkinsfile" target="_blank">example <code>Jenkinsfile</code></a>
+</div>
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/parallel_test_executor/SplitStep/help.html]: Unrelated, but forgotten before.

File: demo/plugins.txt (hunk 1 above)
Comment (abayer) [inline demo/plugins.txt]: Ah, the downside of no longer having one version across the whole suite...

File: src/main/java/org/jenkinsci/plugins/parallel_test_executor/CountDrivenParallelism.java [hunk 3]
```diff
@@ -22,6 +23,7 @@ public int calculate(List<TestClass> tests) {
         return size;
     }
 
+    @Symbol("count")
```
Comment (abayer) [inline src/main/java/org/jenkinsci/plugins/parallel_test_executor/CountDrivenParallelism.java]: Question that just popped to mind - are the `@Symbol`s scoped by anything? Or if we use `count` as the symbol here, does that mean nothing else can use `count` anywhere else? Worrying that we may need some kind of `@Symbol` registry for tracking the things...

File: src/main/java/org/jenkinsci/plugins/parallel_test_executor/MultipleBinaryFileParameterFactory.java [hunk 4]
```diff
@@ -59,7 +58,11 @@ public MultipleBinaryFileParameterFactory(List<ParameterBinding> parametersList)
         for (final ParameterBinding parameterBinding : parametersList) {
             // save them into the master because FileParameterValue might need files after the slave workspace have disappeared/reused
             FilePath target = new FilePath(build.getRootDir()).child("parameter-files");
-                int k = build.getWorkspace().copyRecursiveTo(parameterBinding.filePattern, target);
+            FilePath workspace = build.getWorkspace();
+            if (workspace == null) {
+                throw new AbortException("no workspace");
+            }
+                int k = workspace.copyRecursiveTo(parameterBinding.filePattern, target);
```
Comment (abayer) [inline src/main/java/org/jenkinsci/plugins/parallel_test_executor/MultipleBinaryFileParameterFactory.java]: :ant: Inconsistent indentation?

File: src/main/java/org/jenkinsci/plugins/parallel_test_executor/CountDrivenParallelism.java (hunk 3 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/parallel_test_executor/CountDrivenParallelism.java]: Yes they are scoped by extension point. See Javadoc.

File: src/main/java/org/jenkinsci/plugins/parallel_test_executor/MultipleBinaryFileParameterFactory.java (hunk 4 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/parallel_test_executor/MultipleBinaryFileParameterFactory.java]: Was already like that, just trying to minimize the diff hunk.

File: src/main/java/org/jenkinsci/plugins/parallel_test_executor/CountDrivenParallelism.java (hunk 3 above)
Comment (abayer) [inline src/main/java/org/jenkinsci/plugins/parallel_test_executor/CountDrivenParallelism.java]: That is an immense relief. =)

File: src/main/java/org/jenkinsci/plugins/parallel_test_executor/MultipleBinaryFileParameterFactory.java (hunk 4 above)
Comment (abayer) [inline src/main/java/org/jenkinsci/plugins/parallel_test_executor/MultipleBinaryFileParameterFactory.java]: No worries - just checking.

**Comment by abayer** [comment]
:bee: as is, but I assume there's more commits on the way for the demo.

**Comment by abayer** [comment]
re-:bee:

**Pull request merged**